src/core/MIGRATION.md

```diff
@@ -848,3 +849,46 @@ class MyPlugin {
     this.config$ = initializerContext.config.createIfExists();
   }
 ```
+
+### Mock core services in tests
+Core services already provide mocks to simplify testing and make sure plugins always rely on valid public contracts.
+```typescript
+// my_plugin/server/plugin.test.ts
+import { configServiceMock } from 'src/core/server/mocks.ts'
+
+const configService = configServiceMock.create();
+configService.atPath.mockReturnValue(config$);
+…
+const plugin = new MyPlugin({ configService }, …)
+```
+
+This is not mandatory, but we strongly recommended to export your plugin mocks as well in order for dependent plugins to use them in tests. Your plugin mocks should be exported from the root level of the plugin.
```

Comment:
Can we make this clear that you should only need to export mocks for your setup and start contracts? Not the entire plugin.